SignalServiceKit/src/Loki/Protocol/Sync Messages/SyncMessagesProtocol.swift

```diff
@@ -168,20 +168,21 @@ public final class SyncMessagesProtocol : NSObject {
         // Try to establish sessions
         for hexEncodedPublicKey in hexEncodedPublicKeys {
             let thread = TSContactThread.getOrCreateThread(withContactId: hexEncodedPublicKey, transaction: transaction)
-            let friendRequestStatus = thread.friendRequestStatus
+            let friendRequestStatus = storage.getFriendRequestStatus(for: hexEncodedPublicKey, transaction: transaction)
             switch friendRequestStatus {
-            case .none:
+            case .none, .requestExpired:
                 let messageSender = SSKEnvironment.shared.messageSender
+                // TODO: Does the function below need to handle multi device??
+                // We need to send a FR message here directly to the user. Multi device doesn't come into play.
```

Comment:
Say A1 has a conversation with B1 and B2. A2 is now linked to A1. We'd want A2 to establish sessions with both B1 and B2, right?

Your comment did make me realize that we're not checking the friend request status for the slave device here, and I agree that we should check that to be safe (though theoretically it should never be anything other than `.none` at this point).

Reply:
Naturally `B1` and `B2` will have their pubKeys included in `hexEncodedPublicKeys`, so using multi-device is doing double the work

Reply:
We actually currently exclude slave devices when sending a contact sync message, because of the `thread.shouldThreadBeVisible && !thread.isForceHidden` check

Reply:
Oh ok